bindings/python/gen_docs.py

```diff
@@ -0,0 +1,111 @@
+from os.path import dirname
+from hashlib import md5
+import re
+
+base = dirname(dirname(dirname(__file__)))
+
+# This script runs with no CLI arguments
+# and scrapes documentation comments and c++ function signatures
+# from the manifold sources, in order to generate a c++ header file 
+# exposing the comments as string variables named by function names.
+# This allows python bindings to re-use the c++ comments directly.
+# Some snake-casing of params is applied for python use case.
+
+def snake_case(name):
+    return re.sub("([a-z0-9])([A-Z])", r"\1_\2", name).lower()
+
+
+def param_snake_case(m):
+    return f":{snake_case(m[0][1:])}:"
+
+
+def method_key(name):
+    name = re.sub("\+", "_plus", name)
+    name = re.sub("\-", "_minus", name)
+    name = re.sub("\^", "_xor", name)
+    name = re.sub("\=", "_eq", name)
+    name = re.sub("\:", "_", name)
+    name = re.sub("\~", "destroy_", name)
+    return name
+
+
+parens_re = re.compile(r"[^(]+\(([^(]*(\(.*\))*[^(]*\))", flags=re.DOTALL)
+args_re = re.compile(
+    r"^[^,^\(^\)]*(\(.*\))*[^,^\(^\)]*[\s\&\*]([0-9\w]+)\s*[,\)]", flags=re.DOTALL
+)
+
+
+def parse_args(s):
+    par = parens_re.match(s)
+    if not par:
+        return None
+    out = []
+    arg_str = par[1]
+    while m := re.search(args_re, arg_str):
+        out += [snake_case(m[2])]
+        arg_str = arg_str[m.span()[1] :]
+    return out
+
+
+def collect(fname, matcher):
+    param_re = re.compile(r"@param (\w+)")
+    comment = ""
+    with open(fname) as f:
+        for line in f:
+            line = line.lstrip()
+            if line.startswith("/**"):
+                comment = ""
+            elif line.startswith("*/"):
+                pass
+            elif line.startswith("*") and comment is not None:
+                comment += line[1:].lstrip()
+            elif comment and (m := matcher(line)):
+                while (args := parse_args(line)) is None:
+                    line += next(f)
+                    if len(line) > 500:
+                        break
+
+                method = method_key(snake_case(m[1]))
+                comment = re.sub(param_re, param_snake_case, comment)
+                method = "__".join([method, *args])
+                assert method not in comments
+                comments[method] = comment
+                comment = ""
+
+
+comments = {}
+
+method_re = re.compile(r"(\w+::[\w\-\+\^\=\:]+)\(")
+function_re = re.compile(r"([\w\-\+\^\=\:]+)\(")
+
+
+# we don't handle inline functions in classes properly
+# so instead just white-list functions we want
+def select_functions(s):
```

Comment:
Should we add something in a README somewhere that when a new non-member function is added to the API it needs to also be added to this list?